goship.go

```diff
@@ -760,23 +777,210 @@ func extractOutputHandler(fn func(http.ResponseWriter, *http.Request, string, st
 	}
 }
 
+type User struct {
+	UserName, UserAvatar string
+	auth                 auth
+}
+
+func getUser(r *http.Request) (User, error) {
+	u := User{}
+	session, err := store.Get(r, session_name)
+	if err != nil {
+		return u, errors.New("Error Getting User Session")
+	}
+	if _, ok := session.Values["userName"]; !ok {
+		u.UserName = "genericUSER"
+		u.UserAvatar = "https://camo.githubusercontent.com/33a7d9a138ac73ece82dee977c216eb13dffc984/687474703a2f2f692e696d6775722e636f6d2f524c766b486b612e706e67"
+		return u, errors.New("No username")
+	}
+	if _, ok := session.Values["avatarURL"]; !ok {
+		return u, errors.New("No avatar")
```

Comment:
I'd recommend reading these two nice posts about error handling in Go:
 - http://blog.golang.org/error-handling-and-go
 - https://blog.golang.org/errors-are-values

This is fine, but it makes it hard for functions calling this function to act on the types of errors (which is probably why they're being swallowed); it might be a good idea to define the errors that can be returned in variables, so callers can compare to those and act accordingly. There are other ways of doing it too, as outlined in the posts I linked above